device/mellanox/x86_64-mlnx_lssn2700-r0/plugins/sfputil.py

```diff
@@ -10,6 +10,34 @@
 except ImportError as e:
     raise ImportError("%s - required module not found" % str(e))
 
+def determine_in_docker():
+    with open('/proc/self/cgroup') as f:
+        lines = [line.rstrip('\n') for line in f]
+    for line in lines:
+        words = line.split('/')
+        if len(words) > 1 and words[1] == 'docker':
+            return True
+
+    return False
+
+INSIDE_DOCKER = determine_in_docker()
+
+if INSIDE_DOCKER:
```

Comment:
Why only connect to the Redis DB if running inside a Docker container? If this is to determine whether or not the plugin is being used by xcvrd, it's not an intuitive check. I think it would be better to create a function to connect to the State DB and call that function in `get_transceiver_change_event()`.

Reply:
sfputil plugin is also called by 'show interface transceiver' CLI on the host, but the swsscommon is not available on the host, to make it work both on the host and the docker, need to conditionally import swsscommon.

Reply:
revised.